### PR TITLE
feat(fx-2641): converts medium => additionalGeneIDs filter

### PR DESCRIPTION
**Pull request by dzucconi**
The type of this PR is: **Feature**

This PR resolves [FX-2641]

### Description

So this... works. It just feels obviously incorrect. I want to be able to dispatch a filter select without having to shovel everything into a single value; rather support multiple selected filters on the same param. Right now, if I wanted to merge this I'd have to write a humanization function to turn the selected values into readable labels (currently they are the values; which are slugs). This is, of course, insane. Send help.

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[FX-2641]: https://artsyproduct.atlassian.net/browse/FX-2641

### CODE REVIEW

File: src/lib/Components/FilterModal/FilterModal.tsx [hunk 1]
```diff
@@ -46,20 +47,21 @@ export type FilterScreen =
   | "artistIDs"
   | "artistsIFollow"
   | "attributionClass"
-  | "color"
   | "categories"
+  | "color"
   | "dimensionRange"
   | "estimateRange"
   | "gallery"
+  | "geneIDs"
```
Comment (mzikherman) [inline src/lib/Components/FilterModal/FilterModal.tsx:55]: FYI we actually had to change this, as `geneIDs` param can only be used by gravity in an AND style, whereas I believe the goal here is OR. For the OR style list, its `additionalGeneIDs`: https://github.com/artsy/metaphysics/blob/486eddce74c6ece92e2b4e7ba64d7b1741d8dd58/src/schema/v2/filterArtworksConnection.ts#L102

**Comment by dzucconi** [comment]
Updated geneIDs => additionalGeneIDs. Will have to just update all the `medium` arguments call sites if this looks fine.

File: src/lib/Components/FilteredInfiniteScrollGrid/FilteredInfiniteScrollGrid.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// TODO
```
Comment (dzucconi) [inline src/lib/Components/FilteredInfiniteScrollGrid/FilteredInfiniteScrollGrid.tsx:1]: OOOOK so there's two instances where I have no idea whether or not the `additionalGeneIDs` argument is present on the connection. Not sure what this component is; but it doesn't look like it's used anywhere? Safe to delete?

File: src/lib/Containers/Gene.tsx [hunk 3]
```diff
@@ -1,3 +1,4 @@
+// TODO
```
Comment (dzucconi) [inline src/lib/Containers/Gene.tsx:1]: OK, I *think* this one is the old filter and can be ignored

File: src/lib/utils/ArtworkFilter/FilterArtworksHelpers.ts [hunk 4]
```diff
@@ -170,6 +172,19 @@ export const changedFiltersParams = (currentFilterParams: FilterParams, selected
   return changedFilters
 }
 
+/**
+ * NOTE: I am not happy this exists — but right now we can only dispatch arrays of paramValues
```
Comment (dzucconi) [inline src/lib/utils/ArtworkFilter/FilterArtworksHelpers.ts:176]: 😭

Comment (dblandin) [reply src/lib/utils/ArtworkFilter/FilterArtworksHelpers.ts:176]: 😢 but glad you put an inline comment in 👍

**Review by dblandin — APPROVED** [review]
Sorry that this was such a slog! Looks good to me 👍

**Pull request merged**